api/v1beta1/disruption_webhook.go

```diff
@@ -170,3 +189,135 @@ func (r *Disruption) getMetricsTags() []string {
+		}
+
+		if r.Spec.Network != nil {
+			if caught := safetyNetNeitherHostNorPort(*r); caught {
+				logger.Debugw("The specified disruption either contains no Hosts or contains a Host which has neither a port or a host. The more ambiguous, the larger the blast radius.", r.Name, "SafetyNet Catch", "Network")
+
+				responses = append(responses, "The specified disruption either contains no Hosts or contains a Host which has neither a port or a host. The more ambiguous, the larger the blast radius.")
+			}
+		}
+	}
+
+	return responses, nil
+}
+
+// safetyNetCountNotTooLarge is the safety net regarding the count of targets
+// it will check against the number of targets being targeted and the number of targets in the k8s system
+// > 66% of the k8s system being targeted warrants a safety check if we assume each of our targets are replicated
+// at least twice. > 80% in a namespace also warrants a safety check as namespaces may be shared between services.
+// returning true indicates the safety net caught something
+func safetyNetCountNotTooLarge(r Disruption) (bool, error) {
+	if r.Spec.Unsafemode != nil && r.Spec.Unsafemode.DisableCountTooLarge {
+		return false, nil
+	}
+
+	userCount := r.Spec.Count
+	totalCount := 0
+	namespaceCount := 0
+	namespaceThreshold := 0.8
+	clusterThreshold := 0.66
+
+	if r.Spec.Unsafemode.Config != nil && r.Spec.Unsafemode.Config.CountTooLarge != nil {
+		if r.Spec.Unsafemode.Config.CountTooLarge.NamespaceThreshold != 0 {
+			namespaceThreshold = float64(r.Spec.Unsafemode.Config.CountTooLarge.NamespaceThreshold) / 100.0
+		}
+
+		if r.Spec.Unsafemode.Config.CountTooLarge.ClusterThreshold != 0 {
+			clusterThreshold = float64(r.Spec.Unsafemode.Config.CountTooLarge.ClusterThreshold) / 100.0
+		}
+	}
+
+	if r.Spec.Level == chaostypes.DisruptionLevelPod {
+		pods := &corev1.PodList{}
+		listOptions := &client.ListOptions{
+			Namespace: r.ObjectMeta.Namespace,
+		}
+
+		err := k8sClient.List(context.Background(), pods, listOptions)
+		if err != nil {
+			return false, fmt.Errorf("error listing target pods: %w", err)
+		}
+
+		namespaceCount = len(pods.Items)
+
+		err = k8sClient.List(context.Background(), pods)
+		if err != nil {
+			return false, fmt.Errorf("error listing target pods: %w", err)
+		}
+
+		totalCount = len(pods.Items)
+	} else {
+		nodes := &corev1.NodeList{}
+
+		err := k8sClient.List(context.Background(), nodes)
+		if err != nil {
+			return false, fmt.Errorf("error listing target pods: %w", err)
+		}
+
+		totalCount = len(nodes.Items)
+	}
+
+	userCountVal := float64(userCount.IntVal)
+
+	if userCount.Type != intstr.Int {
+		userCountInt, err := strconv.Atoi(strings.TrimSuffix(userCount.StrVal, "%"))
+		if err != nil {
+			return false, fmt.Errorf("failed to convert percentage to int: %w", err)
+		}
+
+		if namespaceCount != 0 {
+			userCountVal = float64(userCountInt) / 100.0 * float64(namespaceCount)
```

Comment:
It seems to me that if we are getting the count as a %, then there's no need to do the k8s list api calls at all? We can just directly compare the target % to the safe %?

Reply:
It also seems we aren't applying the selectors here, so these `userCountVal`s aren't necessarily accurate?

Reply:
Comparing those two percentages wouldnt accomplish the goal I believe. Target % represents the percentage of targets out of the total number of targets while the Safe % represents the percentage of targets out of either the total number of pods in a namespace or the total number of pods in a cluster. 

 So 20% of pods labeled "app:chaos" could be much different than 20% of the namespace that the targets labeled "app:chaos" live in and of course it would be much different than 20% of the total number of pods on the cluster. So we still need to use the API to see how many total pods are in the namespace. I know there are services that have only the same targets in a namespace, but others which have several services in a single namespace. Let me know if I misunderstood.

Also to note the second comment, yes it would be inaccurate, good catch. I have added another API call to check the number of targets in the cluster that match, that value of userCountVal should be accurate now to the number of targets.